### PR TITLE
Update dependencies and Python version to 3.12

### PR DESCRIPTION
**Pull request by jamesstottmoj**
Merging this PR will update the control panel to run on Python 3.12

## :memo: Summary
This PR resolves 

This PR bumps several dependencies and fixes issues in the tests so that the control panel can run against Python 3.12

The changes in this PR are needed because it updates dependencies and core Python version

## :mag: What should the reviewer concentrate on?
- Feedback on specific parts of the code
- Check side effects, if any

## :technologist: How should the reviewer test these changes?
- Run locally. There should be no noticeable difference between main and this branch

## :books: Documentation status

- [x] No changes to the documentation are required
- [ ] This PR includes all relevant documentation
- [ ] Documentation will be added in the future because ... (see issue #...)

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,38 +1,38 @@
...
+django-structlog==7.1.0
 djangorestframework==3.14.0
 djproxy==2.3.6
 elasticsearch-dsl==7.4.1
-gunicorn==20.1.0
+gunicorn==21.2.0
 Jinja2==3.1.3
-kubernetes==25.3.0
-MarkupSafe==2.1.3
+kubernetes==29.0.0
```
Comment (michaeljcollinsuk) [inline requirements.txt:22]: I'm a little bit unsure about this jump - there was some discussion with the rest of the team about whether we can upgrade when [dependabot suggested it](https://github.com/ministryofjustice/analytics-platform-control-panel/pull/1218). Probably not something that we would catch running locally

**Comment by michaeljcollinsuk** [comment]
Really good start on this. The [Dockerfile](https://github.com/ministryofjustice/analytics-platform-control-panel/blob/main/Dockerfile#L12) will need updating as part of this so that it builds an image using python 3.11, which may go some way to resolving the current fail in CI. I wonder what else we may also need to update in the Dockerfile, there may be implications for helm version it installs, something we can discuss that with some of the other guys in the team.

File: tests/frontend/test_forms.py [hunk 2]
```diff
@@ -411,6 +410,10 @@ def test_update_app_with_custom_connection():
         assert "auth0_nomis_auth0_conn_name" in f.errors
 
 
+def test_pass():
```
Comment (michaeljcollinsuk) [inline tests/frontend/test_forms.py:413]: Can be deleted
```suggestion
```

File: tests/frontend/test_forms.py [hunk 3]
```diff
@@ -411,6 +410,10 @@ def test_update_app_with_custom_connection():
...
+    assert True
+
+
```
Comment (michaeljcollinsuk) [inline tests/frontend/test_forms.py:416]: Can be deleted
```suggestion
```

Comment (jamesstottmoj) [reply tests/frontend/test_forms.py]: Removed

**Comment by michaeljcollinsuk** [comment]
I think the python version mentioned in `running.md` needs to be updated - not sure if the minimum version will now be 3.12, but I presume that 3.8 wont work with this new set of deps?

File: tests/api/models/test_ip_allowlist.py [hunk 4]
```diff
@@ -44,10 +44,10 @@ def test_ip_allowlist_without_app_save_does_not_update_aws_secrets_manager(
 
 # def test_ip_allowlist_delete_updates_aws_secrets_manager(update_aws_secrets_manager):
```
Comment (michaeljcollinsuk) [inline tests/api/models/test_ip_allowlist.py:45]: Might as well delete this test since its commented out?

**Review by michaeljcollinsuk — CHANGES_REQUESTED** [review]
Just one minor change needed in `running.md` about the python version to use, but otherwise great work!

File: tests/api/models/test_ip_allowlist.py (hunk 4 above)
Comment (jamesstottmoj) [reply tests/api/models/test_ip_allowlist.py]: Removed

**Comment by jamesstottmoj** [comment]
Running.md updated

**Review by michaeljcollinsuk — DISMISSED** [review]
🚀

**Review by michaeljcollinsuk — APPROVED** [review]
LGTM

**Comment by michaeljcollinsuk** [comment]
Closing as replaced by https://github.com/ministryofjustice/analytics-platform-control-panel/pull/1285

**Pull request closed without merging**